gr-digital/lib/crc32_bb_impl.cc

```diff
@@ -72,7 +72,7 @@ namespace gr {
         if (n_packed_length > d_buffer.size()){
           d_buffer.resize(n_packed_length);
           }
-        d_buffer.clear();
+        std::fill(d_buffer.begin(), d_buffer.begin() + n_packed_length, 0);
```

Comment:
I don't know! That's why I'm asking you. I'm just zeroing the elements that then get modified in the following `for` loop. If `end` is right here, then `d_buffer.resize()` probably shouldn't be conditional, either.

Reply:
I don't know! That's why I'm asking you. I'm just zeroing the elements that then get modified in the following `for` loop. If `end` is right here, then `d_buffer.resize()` probably shouldn't be conditional, either.

Reply:
I think both ways are ok. The only difference is if the buffer is larger than currently needed only the used space will be filled up by your approach, otherwise everything will be wiped.

Reply:
so, functionally, both is fine? I.e. the elements after `n_packed_length` aren't used at all? Then, let's `resize()`. It doesn't cost anything (and the "now unused" vector space isn't usually freed, as that requires memory management far smarter than what current libCs are doing), but will allow us to catch other bugs, right?